samples/place-autocomplete-data-session/index.ts

```diff
@@ -5,175 +5,145 @@
+}
 
-  // If the request has been superseded by a newer request, do not render the output.
-  if (requestId !== newestRequestId) return;
+async function makeAutocompleteRequest(inputEvent) {
+    // To avoid race conditions, store the request ID and compare after the request.
+    const requestId = ++newestRequestId
+
+    const { AutocompleteSuggestion } = (await google.maps.importLibrary(
+        'places'
+    )) as google.maps.PlacesLibrary;
+
+    // Reset elements and exit if an empty string is received.
+    if (!inputEvent.target ||
+        inputEvent.target.value === '') {
+        titleElement.textContent = '';
+        resultsContainerElement.replaceChildren();
+        return;
+    }
 
-  titleElement.innerText = `Place predictions for "${request.input}"`;
+    // Add the latest char sequence to the request.
+    request.input = (inputEvent.target as HTMLInputElement).value;
 
-  // Clear the list first.
-  resultsContainerElement.replaceChildren();
+    // Fetch autocomplete suggestions and show them in a list.
+    const { suggestions } =
+        await AutocompleteSuggestion.fetchAutocompleteSuggestions(request);
 
-  for (const suggestion of suggestions) {
-    const placePrediction = suggestion.placePrediction;
+    // If the request has been superseded by a newer request, do not render the output.
+    if (requestId !== newestRequestId) return;
 
-    if (!placePrediction) {
-      continue;
-    }
+    titleElement.innerText = `Place predictions for "${request.input}"`;
 
-    // Create a link for the place, add an event handler to fetch the place.
-    const a = document.createElement("a");
-    a.addEventListener("click", () => {
-      onPlaceSelected(placePrediction!.toPlace());
-    });
-    a.innerText = placePrediction!.text.toString();
+    // Clear the list first.
+    resultsContainerElement.replaceChildren();
 
-    // Create a new list item element.
-    const li = document.createElement("li");
-    li.appendChild(a);
-    resultsContainerElement.appendChild(li);
-  }
+    for (const suggestion of suggestions) {
+        const placePrediction = suggestion.placePrediction;
+
+        if (!placePrediction) {
+            continue
+        }
+
+        // Create a link for the place, add an event handler to fetch the place.
+        // We are using a button element to take advantage of its a11y capabilities.
+        const link = document.createElement('button');
+        link.addEventListener('click', () => {
+            onPlaceSelected(placePrediction.toPlace());
+        });
+        link.textContent = placePrediction.text.toString();
+        link.classList.add('link-button');
+
+        // Create a new list item element.
+        const li = document.createElement('li');
+        li.appendChild(link);
+        resultsContainerElement.appendChild(li);
+    }
 }
 
 // Event handler for clicking on a suggested place.
 async function onPlaceSelected(place: google.maps.places.Place) {
```

Comment:
I'm realizing that technically this is also async rendering, so technically it's possible for there to be a race condition where the user types again and makes a new selection. That would warrant its own separate "request id" tracking as well.

that said, in practice that should be pretty darn hard to accomplish, and this is sample code.

your call.